In [1]:
X = 12

if (X > 10 & X < 15):
print('YES')
else:
print('No')

IndentationError: expected an indented block after 'if' statement on line 3 (1825208694.py, line 4)

**Questions 1** 

Maximum of Three Numbers
Complete the piece of code to print a maximum of three variables, a, b and c.

a=10
b=16
c=20

if( # missing part 1 # ):
    print("a")
elif( # missing part 2 #):
    print("b")
else:
     print("c")


Missing Part 1: a > b and a > c           

Missing Part 2: b > a and b > c # Solution

**Question 2**

Reverse Engineering :
Which of the following sequences of inputs would give the output 'C' 'A' 'D' 'B' over multiple iterations?

Given the code below, the output obtained in several runs is 'C' 'A' 'D' 'B'.
if score >= 90:
   print('A')
elif score >=80:
   print('B')
elif score >= 70:
   print('C')
elif score >= 60:
   print('D')
else:
   print('F')

In [14]:
score = 88
if score >= 90:
   print('A')
elif score >=80:
   print('B')
elif score >= 70:
   print('C')
elif score >= 60:
   print('D')
else:
   print('F')

B


In [15]:
if (10 < 0) and (0 < -10):
    print("A")
elif (10 > 0) or False:
    print("B")
else:
    print("C")

B


In [16]:
if True or True:
    if False and True or False:
        print('A')
    elif False and False or True and True:
        print('B')
    else:
        print('C')
else:
    print('D')

B


In [17]:
input_str = "GAARA"

In [18]:
if input_str[0] in "AEIOUaeiou":
    print("YES")
else:
    print("NO")

NO


In [19]:
pin = input("Enter your four digit pin: ")
while pin != '1234':                           #Pre-set pin is 1234
    pin = input('Invalid input, please try again: ')  
print("Pin validation successful.")

Enter your four digit pin: 115
Invalid input, please try again: \654
Invalid input, please try again: 4654
Invalid input, please try again: 5454
Invalid input, please try again: 45354
Invalid input, please try again: 354534
Invalid input, please try again: 1234
Pin validation successful.


In [1]:
import sys    #required for exiting the code and displaying an error

pin = input("Enter your four digit pin: ")
attempt_count = 1
while pin != '1234':
    if attempt_count >= 3:
     sys.exit("Too many invalid attempts")   #error code
    pin = input('Invalid input, please try again: ')  
    attempt_count += 1
print("Pin validation successful.")

Enter your four digit pin: 64654
Invalid input, please try again: 45+55
Invalid input, please try again: +59+5+5+


SystemExit: Too many invalid attempts

C:\Users\GWG1\anaconda3\Lib\site-packages\IPython\core\interactiveshell.py:3534: UserWarning: To exit: use 'exit', 'quit', or Ctrl-D.
  warn("To exit: use 'exit', 'quit', or Ctrl-D.", stacklevel=1)


In [6]:
d = {1:["one",21],2:["one",22],3:["one",23]}

In [11]:
for x in d.keys():print(x)

1
2
3


In [16]:
dy = "Nishanth"

In [20]:
for x in dy:print(x,end="")

Nishanth

In [24]:
L1 = [10, 20, 30, 24, 18]
L2 = [8, 14, 15, 20, 10]
L3=[]
for  i in range(len(L1)):
    L3.append(L2[i] -L1[i])

In [25]:
L3

[-2, -6, -15, -4, -8]

In [ ]:
import ast,sys
input_str = sys.stdin.read()
input_list = ast.literal_eval(input_str)
new = []
for i in input_list:
    new.append(i.title())
print(new)

In [26]:
limit = int(input("Enter the upper limit (positive integer): "))

print("Prime numbers up to", limit, ":")

for num in range(2, limit + 1):
  is_prime = True
  for i in range(2, int(num**0.5) + 1):
    if num % i == 0:
      is_prime = False
      break  # Exit the inner loop if a divisor is found
  if is_prime:
    print(num)


Enter the upper limit (positive integer): 20
Prime numbers up to 20 :
2
3
5
7
11
13
17
19


In [30]:
for i in range(2,20): 
    flag = True
    for n in range(2,i):
        if i%n == 0:
            print(i,"is not prime")
            flag = False
            break        
    if flag == True:print(i, "is prime")
         

2 is prime
3 is prime
4 is not prime
5 is prime
6 is not prime
7 is prime
8 is not prime
9 is not prime
10 is not prime
11 is prime
12 is not prime
13 is prime
14 is not prime
15 is not prime
16 is not prime
17 is prime
18 is not prime
19 is prime


In [31]:

l = []
for i in range(-100, 0):
    if(i % 2 == 0):
        l.append(i)
l

[-100,
 -98,
 -96,
 -94,
 -92,
 -90,
 -88,
 -86,
 -84,
 -82,
 -80,
 -78,
 -76,
 -74,
 -72,
 -70,
 -68,
 -66,
 -64,
 -62,
 -60,
 -58,
 -56,
 -54,
 -52,
 -50,
 -48,
 -46,
 -44,
 -42,
 -40,
 -38,
 -36,
 -34,
 -32,
 -30,
 -28,
 -26,
 -24,
 -22,
 -20,
 -18,
 -16,
 -14,
 -12,
 -10,
 -8,
 -6,
 -4,
 -2]

In [32]:
sorted(set(range(-2, -101, -2)))

[-100,
 -98,
 -96,
 -94,
 -92,
 -90,
 -88,
 -86,
 -84,
 -82,
 -80,
 -78,
 -76,
 -74,
 -72,
 -70,
 -68,
 -66,
 -64,
 -62,
 -60,
 -58,
 -56,
 -54,
 -52,
 -50,
 -48,
 -46,
 -44,
 -42,
 -40,
 -38,
 -36,
 -34,
 -32,
 -30,
 -28,
 -26,
 -24,
 -22,
 -20,
 -18,
 -16,
 -14,
 -12,
 -10,
 -8,
 -6,
 -4,
 -2]

In [33]:
list(range(-100, -1, 2))

[-100,
 -98,
 -96,
 -94,
 -92,
 -90,
 -88,
 -86,
 -84,
 -82,
 -80,
 -78,
 -76,
 -74,
 -72,
 -70,
 -68,
 -66,
 -64,
 -62,
 -60,
 -58,
 -56,
 -54,
 -52,
 -50,
 -48,
 -46,
 -44,
 -42,
 -40,
 -38,
 -36,
 -34,
 -32,
 -30,
 -28,
 -26,
 -24,
 -22,
 -20,
 -18,
 -16,
 -14,
 -12,
 -10,
 -8,
 -6,
 -4,
 -2]

In [34]:
d = {0: 'Fish', 1: 'Bird', 2: 'Mammal'}
for i in d:
    print(i)

0
1
2


In [35]:
updated_dict = {i : i**2 for i in range(2,21) if i % 2 ==0}
print(updated_dict)

{2: 4, 4: 16, 6: 36, 8: 64, 10: 100, 12: 144, 14: 196, 16: 256, 18: 324, 20: 400}


In [ ]:
import ast, sys
n = int(input())

print([i**2 for i in range(1,n+1)])

In [36]:

input_list = ['wood','old','apple','big','item','euphoria']

list_vowel = [input_list[i] for i in range(0,len(input_list)) if input_list[i][0] in "aeiouAEIOU"]

print(list_vowel)

['old', 'apple', 'item', 'euphoria']


In [ ]:
import ast,sys
input_str = sys.stdin.read()
input_list = ast.literal_eval(input_str)

list_vowel = [input_list[i] for i in range(0,len(input_list)) if input_list[i][0] in "aeiouAEIOU"]

print(list_vowel)

In [37]:
output_dict= [val:val**3 for val in input_list if val%3=0]

SyntaxError: invalid syntax (1808746157.py, line 1)

In [38]:
output_dict= {for val in input_list if val%3==0 val:val**3 }

SyntaxError: invalid syntax (3227451013.py, line 1)

In [39]:
output_dict= {val : val**3 for val in input_list if val % 3 == 0}

TypeError: not all arguments converted during string formatting

In [40]:
output_dict= { if val%3==0 val:val**3 for val in input_list}

SyntaxError: invalid syntax (3394465945.py, line 1)

In [41]:
print([i+j for i in "abc" for j in "def"])

['ad', 'ae', 'af', 'bd', 'be', 'bf', 'cd', 'ce', 'cf']


In [42]:
d = {x.upper(): x*3 for x in 'acbd'}
print(d)

{'A': 'aaa', 'C': 'ccc', 'B': 'bbb', 'D': 'ddd'}


In [43]:
def func(name,age = 25,city = "Newyork"):
    print(name,age,city)

In [49]:
func("nishanth",city = "Chennai")

nishanth 25 Chennai


In [50]:
def fun(*args):
    print(args)

In [51]:
fun(54,"gjnsrgnlgrlg",364)

(54, 'gjnsrgnlgrlg', 364)


In [160]:
def func(x, y = 1):
    z = x * y + x + y
    return z

func(2, func(3))

23

In [161]:
def func(x, y):
    z = x * y + x + y
    return z

func(2, func(3))

TypeError: func() missing 1 required positional argument: 'y'

In [162]:
def func(x = 1 ,y = 2):
    z = x * y + x + y
    return z

func(2, func(3))

35

In [167]:
L1 = [2, 7, 8, 10, 3]

def func(y):
    return y**2 - 2 * y - 2

ans_list = [func(x) for x in L1 ]
ans_list

[-2, 33, 46, 78, 1]

In [52]:
min = (lambda x, y: x if x < y else y)
min(101*99, 102*98)


9996

In [53]:
s = ['ad', 'ae', 'af', 'bd', 'be', 'bf', 'cd', 'ce', 'cf']

In [54]:
f = lambda x : x*2

In [56]:
f("ad")

'adad'

In [57]:
map(f,s)

In [67]:
list(map(f,s))

['adad', 'aeae', 'afaf', 'bdbd', 'bebe', 'bfbf', 'cdcd', 'cece', 'cfcf']

In [62]:
f(s)

['ad',
 'ae',
 'af',
 'bd',
 'be',
 'bf',
 'cd',
 'ce',
 'cf',
 'ad',
 'ae',
 'af',
 'bd',
 'be',
 'bf',
 'cd',
 'ce',
 'cf']

In [63]:
len(s)

9

In [64]:
len(f(s))

18

In [71]:
l = [ 1,2,3,4,5,6,7,8,9,10]

In [72]:
list(map(lambda x : x**2,l))

[1, 4, 9, 16, 25, 36, 49, 64, 81, 100]

In [73]:
x = [1, 4, 9, 16, 25, 36, 49, 64, 81, 100]

In [74]:
s = lambda x : x%2 ==0

In [78]:
list(filter(s,x))

[4, 16, 36, 64, 100]

In [93]:
dy = {'person1': ['Alice', 25],
    'person2': ['Bob', 30],
    'person3': ['Charlie', 35],
    'person4': ['David', 40],
    'person5': ['Eve', 45],
    'person6': ['Frank', 50],
    'person7': ['Grace', 55],
    'person8': ['Helen', 60],
    'person9': ['Ian', 65],
    'person10': ['Jack', 70]
}

In [94]:
dy

{'person1': ['Alice', 25],
 'person2': ['Bob', 30],
 'person3': ['Charlie', 35],
 'person4': ['David', 40],
 'person5': ['Eve', 45],
 'person6': ['Frank', 50],
 'person7': ['Grace', 55],
 'person8': ['Helen', 60],
 'person9': ['Ian', 65],
 'person10': ['Jack', 70]}

In [95]:
y = lambda x : x[1] > 40 

In [97]:
y(['Nishanth',35])

False

In [98]:
list(filter(y,dy.values()))

[['Eve', 45],
 ['Frank', 50],
 ['Grace', 55],
 ['Helen', 60],
 ['Ian', 65],
 ['Jack', 70]]

In [106]:
from functools import reduce

In [107]:
l = [ 1,2,3,4,5,6,7,8,9,10]

In [108]:
f = lambda x,y : x+y

In [109]:
f(2,0)

2

In [110]:
x = reduce(f,l)

In [111]:
x

55

In [112]:
from functools import reduce
list_1 = ['Paul','Ted']
reduce(lambda x,y: x + y,list_1)

'PaulTed'

In [115]:
x = input("Enter a word in small caps:")
for i in x:
    count=0
    if i == "r":
        count =1
        break
if count == 1:
    print("The word belongs to red team")   

Enter a word in small caps:fjsefkjr
The word belongs to red team


In [124]:
x = input("Enter a word in small caps:")
for i in x:
    red = 0
    blue = 0
    no = 0
    if i == "r":
        print("The word belongs to red team")
        red = 1
        break
    elif i =="b":
        print("Blue team")
        blue = 1
        break
if red== 0 and blue ==0:
    print("No Team")

Enter a word in small caps:ahefbhffbsfbshfbshf
Blue team


In [131]:
n = int(input("enter the upper limit:"))

for i in range(1,n+1):
    if i % 17 ==0:
        print(i)

enter the upper limit:50
17
34


In [132]:
sample_list = [6, 98, 34, 45, 64, 7, 2, 5, 78, 90, 324, 111, 657, 438]

list(map(lambda x :x+13,sample_list))


[19, 111, 47, 58, 77, 20, 15, 18, 91, 103, 337, 124, 670, 451]

In [134]:
[x + 13 for x in sample_list]

[19, 111, 47, 58, 77, 20, 15, 18, 91, 103, 337, 124, 670, 451]

In [138]:
[x + 13 for x in sample_list]

[19, 111, 47, 58, 77, 20, 15, 18, 91, 103, 337, 124, 670, 451]

In [139]:
sample_list = [6, 98, 34, 45, 64, 7, 2, 5, 78, 90, 324, 111, 657, 438]

solu_dict = {}
for num in sample_list:
    fact_list = []
    for i in range (1,num+1):
        if num%i ==0:
            fact_list.append(i)
    solu_dict[num]=fact_list      
solu_dict

{6: [1, 2, 3, 6],
 98: [1, 2, 7, 14, 49, 98],
 34: [1, 2, 17, 34],
 45: [1, 3, 5, 9, 15, 45],
 64: [1, 2, 4, 8, 16, 32, 64],
 7: [1, 7],
 2: [1, 2],
 5: [1, 5],
 78: [1, 2, 3, 6, 13, 26, 39, 78],
 90: [1, 2, 3, 5, 6, 9, 10, 15, 18, 30, 45, 90],
 324: [1, 2, 3, 4, 6, 9, 12, 18, 27, 36, 54, 81, 108, 162, 324],
 111: [1, 3, 37, 111],
 657: [1, 3, 9, 73, 219, 657],
 438: [1, 2, 3, 6, 73, 146, 219, 438]}

In [140]:
sample_sentence = "Pack my box with five dozen liquor jugs"

unique_letter = set(sample_sentence)
number = len(unique_letter)
print(unique_letter)
number

{'w', 'f', 'l', 'k', 'q', 'x', 'y', 'n', 'a', 'c', 'r', 'b', 'v', 'i', 'j', 'P', 'z', ' ', 'm', 'o', 'd', 't', 's', 'e', 'u', 'g', 'h'}


27

In [141]:
def calc(operation, x,y):
    if operation == "add":
        return x+y
    if operation == "subs":
        return x-y
    if operation == "multi":
        return x*y
    if operation == "div":
        return x/y
    else:
        print("Enter a valid operation")
        return None
    
print ("Enter the operation you want to execute")
oper = input("The available operations are 'add', 'subs', 'multi', 'div' " )
x = int (input ("Enter the first number"))
y = int (input ("Enter the second number"))

print("Answer is ", calc(oper, x, y))

Enter the operation you want to execute
The available operations are 'add', 'subs', 'multi', 'div' subs
Enter the first number15
Enter the second number10
Answer is  5


In [145]:
num = input("Enter three numbers :")

num_list = num.split()

a = int(num_list[0])
b = int(num_list[1])
c = int(num_list[2])

print (a,b,c)

if a>b: 
    if a>c:
        print ("The largest number is ", a)
    else: 
        print ("The largest number is ", c)
else:
    if b>c:
        print ("The largest number is ", b)
    else: 
        print ("The largest number is ", c)

Enter three numbers :15 9 100
15 9 100
The largest number is  100


In [152]:
num = int(input("Enter the number you want to check:"))
new_list =[]
for i in range(1,num):
    if num % i == 0:
        new_list.append(i)
if sum(new_list) == num:
    print("The number is a perfect number")
else:
    print("The number is not a perfect number")
    

Enter the number you want to check:6
The number is a perfect number


In [154]:
list1 = [5, 8, 9, 12]
list2 = [10, 12, 8, 6]

# write code here 
list(map(lambda x,y : x+y,list1,list2))

[15, 20, 17, 18]

In [158]:
from functools import reduce
num = int(input("Enter a Natural Number:"))

x = reduce(lambda i,y: i * y, range(1,num+1))
print(x)

Enter a Natural Number:5
120


Using the function Map, count the number of words that start with ‘S’ in input_list.



Sample Input:

['Santa Cruz','Santa fe','Mumbai','Delhi']

In [ ]:
import ast,sys
input_str = sys.stdin.read()
input_list = ast.literal_eval(input_str)

count = sum(list(map(lambda x:x[0] == "S",input_list)))

print(count)

Map Function
Description
Create a list ‘name’ consisting of the combination of the first name and the second name from list 1 and 2 respectively. 

For e.g. if the input list is:
[ ['Ankur', 'Avik', 'Kiran', 'Nitin'], ['Narang', 'Sarkar', 'R', 'Sareen']]

the output list should be the list:
['Ankur Narang', 'Avik Sarkar', 'Kiran R', 'Nitin Sareen']

Note: Add a space between first name and last name.

In [ ]:
import ast,sys
input_str = sys.stdin.read()
input_list = ast.literal_eval(input_str)
first_name = input_list[0]
last_name = input_list[1]

name = list(map(lambda x , y : x +" " + y,first_name,last_name))

print(name)

Filter Function
Description
Extract a list of numbers that are multiples of 5 from a list of integers named input_list.



Sample Input:

[1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50]



Sample Output:

[5, 10, 15, 20, 25, 30, 35, 40, 45, 50]



Note: Use the filter() function.

In [ ]:
import ast,sys
input_str = sys.stdin.read()
input_list = ast.literal_eval(input_str)

list_answer = list(filter(lambda x : x%5==0,input_list))

print(list_answer)

Filter Function
Description
You are given a list of strings such as input_list = ['hdjk', 'salsap', 'sherpa'].



Extract a list of names that start with an ‘s’ and end with a ‘p’ (both 's' and 'p' are lowercase) in input_list.



Sample Input:

['soap','sharp','shy','silent','ship','summer','sheep']



Sample Output:

['soap', 'sharp', 'ship', 'sheep']



Note: Use the filter() function.

In [ ]:
import ast,sys
input_str = sys.stdin.read()
input_list = ast.literal_eval(input_str)

sp = list(filter(lambda x : x[0]=='s' and x[-1] == 'p',input_list))

print(sp)

Reduce Function
Description
Using the Reduce function, concatenate a list of words in input_list, and print the output as a string.

If input_list = ['I','Love','Python'], the output should be the string 'I Love Python'.



Sample Input:

['All','you','have','to','fear','is','fear','itself']



Sample Output:

All you have to fear is fear itself

In [ ]:
import ast,sys
input_str = sys.stdin.read()
input_list = ast.literal_eval(input_str)

from functools import reduce

ans = reduce(lambda x,y : x + " " + y,input_list)

print(ans)

Reduce Function
Description
You are given a list of numbers such as input_list = [31, 63, 76, 89]. Find and print the largest number in input_list using the reduce() function.



Sample Input:

[65,76,87,23,12,90,99]



Sample Output:

99

In [ ]:
import ast,sys
input_str = sys.stdin.read()
input_list = ast.literal_eval(input_str)
from functools import reduce

answer = reduce(lambda x,y : x if x > y else y , input_list)

print(answer)

# Control Structures and Functions MCQ

In [168]:
input_string = "I love Python"
k = [(i.upper(), len(i)) for i in input_string.split(' ')]
print(k)

[('I', 1), ('LOVE', 4), ('PYTHON', 6)]


In [169]:
var = [i**2 for i in range(5)]
print(var)

[0, 1, 4, 9, 16]


In [170]:
l1=[1,2,3,7]
l2=[4,5,6,9]
[x+y for x in l1 for y in l2]

[5, 6, 7, 10, 6, 7, 8, 11, 7, 8, 9, 12, 11, 12, 13, 16]

In [171]:
def function(n):
  n=n+1
 
n=2
print(function(2))

None


In [172]:
x=10
def func(x):
   x=20
print(x)
 
print(x)

10
10


In [173]:
def samplefunc(msg,i=3):
  print(msg*i)
 
samplefunc('python',4)
samplefunc('program')

pythonpythonpythonpython
programprogramprogram


In [174]:
f=lambda x:bool(x%2)
print(f(100)* f(101))

0


In [175]:
l=(1, -2, -3, 4, 5)
def f1(x):
    return x//2
m1=map(f1, l)
print(list(m1))

[0, -1, -2, 2, 2]


In [176]:
from functools import reduce
numbers = [-1, 2, -5, 3, 8,-2, 1,-4,-2]
sum = reduce(lambda x, y: x + y, numbers)
print(sum)

0


In [177]:
sample=[-4,2,5,-3]
ans=map(lambda x: x*2, sample)
print(ans)

In [178]:
from functools import reduce
print(list(reduce(lambda x,y:x+y,range(5,15))))

TypeError: 'int' object is not iterable

In [179]:
import functools
lis=[1, 2, 3, 4, 5]
m=functools.reduce(lambda x, y:x if x<y else y, lis)
print(m)

1


In [180]:
lis=[1, 2, 3, 4, 5]
c=lis[0]
for i in range(len(lis)-1):
    if lis[i]<c:
       c=lis[i]
print(c)

1


In [181]:
letters = ('d','i','a','s')
ans = list(map(lambda word: f"{word}x", letters))
print(ans)

['dx', 'ix', 'ax', 'sx']


In [182]:
def sf(a):
    return a%4==0 and a%8!=0
m=filter(sf, range(10, 31))
print(list(m))

[12, 20, 28]


# Practise Questions - I

Even numbers in a list
Description
Given a list of integers, write a python code to find all the even numbers present in the list.



Note: Try using lambda and filter functions to solve the problem.



Input: A list of integers

Output: A list of integers



----------------------------------------------------------------------



Sample input: [5, 7, 22, 97, 54, 62, 77, 23, 73, 61] 

Sample output: [22, 54, 62]



----------------------------------------------------------------------



Sample input: [500, 789, 2112, 9097, 5894, 9062, 8977, 1223, 773, 600]

Sample output: [500, 2112, 5894, 9062, 600]

In [ ]:
import ast,sys
input_str = sys.stdin.read()
input_list = ast.literal_eval(input_str)

ans = list(filter(lambda x : x % 2 == 0,input_list))
print(ans)

Triple the list elements
Description
Given a list of integers, write a python code to triple all the values in the list.



Note: Try using lambda and map functions to solve the problem.



Input: A list of integers

Output: A list of integers



----------------------------------------------------------------------



Sample input: [5, 7, 22, 97, 54, 62, 77, 23, 73, 61]

Output: [15, 21, 66, 291, 162, 186, 231, 69, 219, 183]



----------------------------------------------------------------------



Sample input: [15, 71, 2, 9, 5, 6, 11, 3, 7, 6]

Output: [45, 213, 6, 27, 15, 18, 33, 9, 21, 18]



----------------------------------------------------------------------

In [ ]:
import ast,sys
input_str = sys.stdin.read()
input_list = ast.literal_eval(input_str)

ans = list(map(lambda x : x*3,input_list))
print(ans)

Encrypt elements in a list
Description
A company stores the names of their employees in a list. They want to encrypt the names so that no one can read them and the data stays safe. One of the steps in encrypting is to reverse each name in the list and convert them to uppercase and your task is to write a Python code to execute the same.



Note: Use lambda and map functions.



Input: A list of names

Output: A list of encrypted names



----------------------------------------------------------------------



Sample input: ['Ronaldo', 'Cristiano', 'Rakesh', 'Ronak']

Sample output: ['ODLANOR', 'ONAITSIRC', 'HSEKAR', 'KANOR']



----------------------------------------------------------------------



Sample input: ['Abhay', 'Aravind', 'Desilva', 'Darpan']

Sample output: ['YAHBA', 'DNIVARA', 'AVLISED', 'NAPRAD']



----------------------------------------------------------------------

In [ ]:
import ast,sys
input_str = sys.stdin.read()
input_list = ast.literal_eval(input_str)
ans = list(map(lambda x : x[::-1].upper(),input_list))
print(ans)

# Practise Questions - II

Sorting a dictionary
Description
Given is a list consisting of dictionary elements. Each dictionary contains the name of a student as a first item and his/her corresponding rank as the second item. Your task is to write a Python program to print the list (dictionary elements) in a sorted ascending order according to student ranks. In case the rank of two or more students are the same, then sort them in ascending order (alphabetically) according to their names.



Note: Use the sorted() function to solve the question.



Input: Dictionary

Output: Dictionary



----------------------------------------------------------------------



Sample input: [{ "name" : "Arpit", "rank" : 20}, { "name" : "Manjeet", "rank" : 20 },{ "name" : "Aravind" , "rank" : 19 }]

Sample output: [{ "name" : "Aravind" , "rank" : 19 }, { "name" : "Arpit", "rank" : 20}, { "name" : "Manjeet", "rank" : 20 },

]



----------------------------------------------------------------------



Sample input: [{ "name" : "Mohan", "rank" : 2}, { "name" : "Mandeep", "rank" : 2 },

{ "name" : "Merut" , "rank" : 2 }]



Sample output: [{'name': 'Mandeep', 'rank': 2}, {'name': 'Merut', 'rank': 2}, {'name': 'Mohan', 'rank': 2}]

In [ ]:
import ast,sys
input_str = sys.stdin.read()
input_list = ast.literal_eval(input_str)
ans = sorted(input_list,key = (lambda x:(x["rank"],x["name"])))
print(ans)

Intersection of lists
Description
In ‘ABC’ university, company ‘XYZ’ conducted three rounds of interviews to recruit students. Those students who passed all the rounds were selected. The company released the results for each round in a separate list. Your task is to write a Python code to display the list of the students who passed all the three rounds.



Note: a. Name of the person can be given in uppercase or lowercase.



 b. Print the names in lowercase in the output.



Input: List

List

List



Output: List



----------------------------------------------------------------------



Sample input: ['Arkam', 'Bairstow', 'Cairy', 'Darpan'] 

 ['ARKAM', 'Bairstow', 'Cairy', 'Darpan', 'Dhoni', 'Sachin']

 ['arkam', 'bairstow', 'Cheteshwar', 'Dinesh']

        

Sample output: ['arkam', 'bairstow']



----------------------------------------------------------------------



Sample input: ['Pranay', 'Aditya', 'Deepesh', 'Sandesh']

['Pranoy', 'Surya', 'Cairy', 'Vignesh', 'ADITYA', 'SANDesh']

['Vardhan', 'Shailesh', 'aditya', 'sandesh']

        

Sample output: ['aditya', 'sandesh']



----------------------------------------------------------------------

Sample input: ['Siddharth', 'Gaurav', 'Prasad', 'Kuldeep', 'karna']

['siddharth', 'Gaurav', 'Prasad', 'Kuldeep', 'karna']

['Siddharth', 'gaurav', 'PRASAD', 'KuLDEEp', 'KARNA']



Sample output: ['siddharth', 'gaurav', 'prasad', 'kuldeep', 'karna']

In [ ]:
import ast
list1 = ast.literal_eval(input())
list2 = ast.literal_eval(input())
list3 = ast.literal_eval(input())

round1 = list(map(lambda x: x.lower(),list1))
round2 = list(map(lambda x: x.lower(),list2))
round3 = list(map(lambda x: x.lower(),list3))

ans  = [i.lower() for i in round1 if (i in round2) and (i in round3)]
print(ans)

Sorting a list of dictionary elements
Description
A teacher decided to divide the class into groups to conduct fun activities. Each group had one or more members in it. The data is provided in the form of a list with dictionary elements. The key and value in each dictionary correspond to the student’s roll number and his/her name.



In case the length of two or more dictionaries are the same, then the dictionary that occurs first in the input list will also appear first among the other dictionaries of the same length.



Your task is to write a Python program to sort the list in descending order with respect to the group strength.



Note: Use lambda and sorted functions to solve the question.



Input: List

Output: List



----------------------------------------------------------------------



Sample input: [{4: 'Viking', 9: 'Happy', 10: 'Tony', 2: 'Rogers'}, {12: 'Vikram', 19: 'Vipan'}, {3: 'Kohli'}, {1: 'Ronaldo', 29: 'Harish', 7: 'Ganesh'}]



Sample output: [{4: 'Viking', 9: 'Happy', 10: 'Tony', 2: 'Rogers'}, {1: 'Ronaldo', 29: 'Harish', 7: 'Ganesh'}, {12: 'Vikram', 19: 'Vipan'}, {3: 'Kohli'}]



----------------------------------------------------------------------



Sample input: [{14: 'Darpan', 19: 'Harry', 10: 'Julia'}, {12: 'Ethan', 19: 'Rogers'}, {3: 'Kohli', 1: 'Neymar', 29: 'Sachin', 7: 'Ramesh'}]



Sample output: [{3: 'Kohli', 1: 'Neymar', 29: 'Sachin', 7: 'Ramesh'}, {14: 'Darpan', 19: 'Harry', 10: 'Julia'}, {12: 'Ethan', 19: 'Rogers'}]

In [ ]:
import ast
input_list = ast.literal_eval(input())
sorted_list = sorted(input_list, key=lambda x:len(x),reverse=True)
print(sorted_list)

 END 